GDJS/Runtime/jsonmanager.ts

```diff
@@ -60,40 +65,42 @@ namespace gdjs {
      * as JSON files can have been modified without the editor knowing).
      *
      * @param onProgress The function called after each json is loaded.
-     * @param onComplete The function called when all jsons are loaded.
      */
-    preloadJsons(
-      onProgress: JsonManagerOnProgressCallback,
-      onComplete: JsonManagerOnCompleteCallback
-    ): void {
-      const resources = this._resources;
-      const jsonResources = resources.filter(function (resource) {
-        return (
-          (resource.kind === 'json' ||
-            resource.kind === 'tilemap' ||
-            resource.kind === 'tileset') &&
-          !resource.disablePreload
-        );
-      });
-      if (jsonResources.length === 0) {
-        return onComplete(jsonResources.length);
-      }
-      let loaded = 0;
+    async preloadJsons(
+      onProgress: (loadedCount: integer, totalCount: integer) => void
+    ): Promise<integer> {
+      const preloadedResources = [
+        ...gdjs.filterIterable(
+          this._resources.values(),
+          (resource) => !resource.disablePreload
+        ),
+      ];
```

Comment:
Could you do?

```suggestion
      const preloadedResources = [...this._resources.values()](
          .filter((resource) => !resource.disablePreload);
```

Reply:
Maybe I micro optimized it.

Reply:
Thanks for adapting this 👍 Let's keep things with simple and stupid arrays/maps :)